Mathlib/LinearAlgebra/RootSystem/Reduced.lean

```diff
@@ -68,34 +68,39 @@ lemma IsReduced.linearIndependent_iff [Nontrivial R] [P.IsReduced] :
   · rw [h h']
     exact ⟨1, 1, by simp⟩
 
-lemma two_smul_notMem_range_root [NeZero (2 : R)] [NoZeroSMulDivisors ℤ M] [P.IsReduced] {i : ι} :
-    (2 : R) • P.root i ∉ range P.root := by
-  have _i : Nontrivial R := ⟨2, 0, two_ne_zero⟩
-  have : ¬ LinearIndependent R ![(2 : R) • P.root i, P.root i] := by
-    simpa only [LinearIndependent.pair_iff, not_forall] using ⟨1, -2, by simp, by simp⟩
+lemma nsmul_notMem_range_root [CharZero R] [NoZeroSMulDivisors ℤ M] [P.IsReduced]
+    {n : ℕ} [n.AtLeastTwo] {i : ι} :
+    n • P.root i ∉ range P.root := by
+  have : ¬ LinearIndependent R ![n • P.root i, P.root i] := by
+    simpa only [LinearIndependent.pair_iff, not_forall] using
+      ⟨1, -(n : R), by simp [Nat.cast_smul_eq_nsmul], by simp⟩
   rintro ⟨j, hj⟩
   replace this : j = i ∨ P.root j = -P.root i := by
     simpa only [← hj, IsReduced.linearIndependent_iff, not_and_or, not_not] using this
   rcases this with rfl | this
-  · simp [two_smul, P.ne_zero j] at hj
-  · rw [← one_smul ℤ (P.root i), ← neg_smul] at this
-    rw [← Int.cast_ofNat, Int.cast_smul_eq_zsmul (R := R), this,
-      (smul_left_injective ℤ <| P.ne_zero i).eq_iff] at hj
-    norm_num at hj
-
+  · replace hj : (1 : ℤ) • P.root j = (n : ℤ) • P.root j := by simpa
+    rw [(smul_left_injective ℤ <| P.ne_zero j).eq_iff, eq_comm] at hj
+    have : 2 ≤ n := Nat.AtLeastTwo.prop
+    omega
+  · rw [← one_smul ℤ (P.root i), ← neg_smul, hj] at this
+    replace this : (n : ℤ) • P.root i = -1 • P.root i := by simpa
+    rw [(smul_left_injective ℤ <| P.ne_zero i).eq_iff] at this
+    omega
+
+@[deprecated (since := "2025-07-06")] alias two_smul_notMem_range_root := nsmul_notMem_range_root
 @[deprecated (since := "2025-05-24")] alias two_smul_nmem_range_root := two_smul_notMem_range_root
 
 lemma linearIndependent_of_add_mem_range_root
-    [NeZero (2 : R)] [NoZeroSMulDivisors ℤ M] [P.IsReduced] {i j : ι}
+    [CharZero R] [NoZeroSMulDivisors ℤ M] [P.IsReduced] {i j : ι}
```

Comment:
Note that this isn't really the weakening it might appear to be since:
```lean
example [Nontrivial M] [NoZeroSMulDivisors ℤ M] : CharZero R := CharZero.of_noZeroSMulDivisors R M
```

Of course this raises the issue of demanding `[Nontrivial M]` rather than `[CharZero R]` but I contend that this should happen separately as I found there were quite a few ripple effects from implementing that here.